### DISCUSSION TITLE
Discrepancy between Playground and prod

### DISCUSSION THREAD

**Discussion by l3112** [open-policy-agent/community#436] [OPA and Rego] [answered]
When I format my rego code [here in the Rego Playground](https://play.openpolicyagent.org/p/qjRKWv2Uv2), it puts an `if` in places;

`yes_bucket **if** {
	regex.match("ca.*", bucketName)
}`

`deny_bucket_creation **if** {
	input.request.operation == "CreateBucket"
	not startswith(input.parameters.bucketName, "ca_")
}`

But when I try to run the code with Trivy on my set up, I get 
`rego_parse_error: var cannot be used for rule name`

When I remove it, there's another error that's not relevant here, but I too thought `if` statements was how the language operates -- "allow if" or "don't allow if". Who has the discrepancy here? is `if` implied and works without stating it?

**Comment by anderseknert** [comment, marked as answer]
Hi Morgan! 

The `if` keywords is optional, but recommended for readability. It does however require the use of `import future.keywords.if` in the policy, as the keyword was added fairly recently. I can't speak for, or support, Trivy as it is an external project, but do you include such an import in your policy there?

**Reply by l3112** [reply]
I forget about `import future.keywords` and its ilk all the time, thanks!
